Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,48 @@
+
+## Problem
+
+When users unintentionally hydrate too many files within their enlistment, commands like checkout and commit increase their running time by several orders of magnitude. This causes unacceptable losses in performance and greatly hinders users as they go about their daily development cycle. The primary metric of the overhydration is the number of placeholder files which have been pulled from the remote into the local system. Slowdowns may also stem from having many modified files or directories which increases the amount of work that git does under the hood. Additionally, if a local enlistment has been in use for a long time, the number of loose object may eventually reach the order of several thousand in which case that also contributes to the loss in performance.
+
+## Provided Statistics
+
+### Safe to access
+
+- **Modified paths** - The number of files which have been edited and have been handed over to git
+- **Git index** - The differences between modified paths and the index tracked by git
+- **Placeholders** - A total number of the placeholders being tracked in VFS4G and a list of the placeholders in the system (Note that this contains some upper level directories and thus not all of their children are included in the count without enumeration)
+- **Troublesome programs** - Similar to what is already being logged in the heartbeat, look at which services or programs are responsible for hydrating large amounts of files in the enlistment
+
+### Potentially detrimental to access on a mounted enlistment
+
+- **Placeholder details** - Crawling the file system to enumerate directories of placeholders may unintentionally further hydrate files (IE downloading file sizes), but doing so would allow for greater details such as the percent hydration of folders, the total number of hydrated files, the type of files which are most hydrated, etc.
+- **File system** - Provide enumerated details for files in the placeholder list or the modified path list
+
+## Background Information
+
+- **Modified paths** is the list of files which VFS4G has completely pulled from the remote and whose contents are on disk in their entirety. These files are handed to git via a hook when any git command requiring the list of changed files is called
+- **Placeholders** are files which have been opened by the local system with a read tag but without a write tag. They are objects tracked by VFS4G since git should not have to internally scan them since it is guaranteed that they have not been modified and accordingly do not have to be rehashed for differences. Their contents are on disk, but control over them has not been fully ceded over to the system. They decrease performance because VFS4G has to track all of them individually, and this can cause certain `gvfs` commands to be slow

```

Comment:
>control over them has not been fully ceded over to the system.

over to git?

>this can cause certain `gvfs` commands to be slow

`git` commands that need to update the projection